packages/eslint-plugin-pf-codemods/lib/helpers.js

```diff
@@ -35,12 +35,22 @@ function moveSpecifiers(
       if (firstSpecifier?.parent?.source?.value?.includes("dist/esm")) {
         //expecting @patternfly/{package}/{designator} where designator is next/deprecated
         const toParts = toPackage.split("/");
-        //expecting @patternfly/{package}/dist/esm/components/{Component}/index.js
-        //needing toPath to look like fromPath with the designator before /components
         const fromParts = firstSpecifier.parent.source.value.split("/");
-        if (toParts[0] === "@patternfly" && toParts.length === 3) {
-          fromParts.splice(4, 0, toParts[2]);
-          modifiedToPackage = fromParts.join("/");
+        if (toParts[0] === "@patternfly") {
+          //expecting @patternfly/{package}/dist/esm/components/{Component}/index.js
+          //needing toPath to look like fromPath with the designator before /components
+          if (toParts.length === 3) {
+            fromParts.splice(4, 0, toParts[2]);
+            modifiedToPackage = fromParts.join("/");
+          }
+
+          // Expecting @patternfly/{package}/dist/esm/next/components/{Component}/index.js
+          // Needing toPath to look like fromPath *without* the designator before /components
+          if (toParts.length === 2) {
+            modifiedToPackage = fromParts
+              .filter((part) => part !== "next")
+              .join("/");
+          }
```

Comment:
I wouldn't block over it, but I wonder if this logic could be reworked a bit so that we can avoid getting this deep in nested if statements. IMO nested if statements are (at least most of the time) the kind of thing that increases cognitive load when trying to reason about code exponentially with each level.

Simplest way to do it in this case would probably be to just refactor to have guard clauses for the preceding checks. But there very well could be a better way than that.

Reply:
Yeah I agree, I'll rework this this afternoon since it should be simple enough of a fix.